synapse/lib/nexus.py

```diff
@@ -497,27 +496,32 @@ async def _onTeleLink(self, proxy):
 
     async def runMirrorLoop(self, proxy):
 
-        cellinfo = await proxy.getCellInfo()
-        features = cellinfo.get('features', {})
-        if features.get('dynmirror'):
-            await proxy.readyToMirror()
+        try:
+            cellinfo = await proxy.getCellInfo()
+            features = cellinfo.get('features', {})
+            if features.get('dynmirror'):
+                await proxy.readyToMirror()
+
+            cellvers = cellinfo['cell']['version']
+            if cellvers > self.cell.VERSION:
+                logger.error('Leader is a higher version than we are. Mirrors must be updated first. Entering read-only mode.')
+                await self.addWriteHold(leaderversion)
+                # this will fire again on reconnect...
+                return
 
-        cellvers = cellinfo['cell']['version']
-        if cellvers > self.cell.VERSION:
-            logger.error('Leader is a higher version than we are. Mirrors must be updated first. Entering read-only mode.')
-            await self.addWriteHold(leaderversion)
-            # this will fire again on reconnect...
-            return
+            # When we reconnect and the leader version has become ok...
+            await self.delWriteHold(leaderversion)
 
-        # When we reconnect and the leader version has become ok...
-        await self.delWriteHold(leaderversion)
+            if self.celliden is not None:
+                if self.celliden != await proxy.getCellIden():
+                    logger.error('remote cell has different iden!  Aborting mirror sync')
+                    await proxy.fini()
+                    await self.fini()
+                    return
 
-        if self.celliden is not None:
-            if self.celliden != await proxy.getCellIden():
-                logger.error('remote cell has different iden!  Aborting mirror sync')
-                await proxy.fini()
-                await self.fini()
-                return
+        except Exception as exc:
+            logger.exception('Unknown error during mirror loop startup', exc_info=exc)
```

Comment:
logger.exception will automatically pull the `exc_info` for you when called in a error handler.
```
logger.exception('Error during mirror loop startup: {exc}`)
```